kedro-datasets/tests/spark/test_spark_streaming_dataset.py

```diff
@@ -0,0 +1,134 @@
+import json
+import re
+from pathlib import Path
+
+import pytest
+from kedro.io.core import DataSetError
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, StringType, StructField, StructType
+from pyspark.sql.utils import AnalysisException
+
+from kedro_datasets.spark.spark_dataset import SparkDataSet
+from kedro_datasets.spark.spark_streaming_dataset import SparkStreamingDataSet
+
+SCHEMA_FILE_NAME = "schema.json"
+
+
+def sample_schema(schema_path):
+    with open(schema_path, encoding="utf-8") as f:
+        try:
+            return StructType.fromJson(json.loads(f.read()))
+        except Exception as exc:
+            raise DataSetError(
+                f"Contents of 'schema.filepath' ({schema_path}) are invalid. "
+                f"Schema is required for streaming data load, Please provide a valid schema_path."
+            ) from exc
+
+
+@pytest.fixture
+def sample_spark_streaming_df(tmp_path):
+    schema = StructType(
+        [
+            StructField("sku", StringType(), True),
+            StructField("new_stock", IntegerType(), True),
+        ]
+    )
+    data = [("0001", 2), ("0001", 7), ("0002", 4)]
+    schema_path = (tmp_path / SCHEMA_FILE_NAME).as_posix()
+    with open(schema_path, "w", encoding="utf-8") as f:
+        json.dump(schema.jsonValue(), f)
+    return SparkSession.builder.getOrCreate().createDataFrame(data, schema)
+
+
+class TestStreamingDataSet:
+    def test_load(self, tmp_path, sample_spark_streaming_df):
+        filepath = (tmp_path / "test_streams").as_posix()
+        schema_path = (tmp_path / SCHEMA_FILE_NAME).as_posix()
+
+        spark_json_ds = SparkDataSet(
+            filepath=filepath, file_format="json", save_args=[{"mode", "overwrite"}]
+        )
+        spark_json_ds.save(sample_spark_streaming_df)
+
+        streaming_ds = SparkStreamingDataSet(
+            filepath=filepath,
+            file_format="json",
+            load_args={"schema": {"filepath": schema_path}},
+        ).load()
+        assert streaming_ds.isStreaming
+        schema = sample_schema(schema_path)
+        assert streaming_ds.schema == schema
+
+    def test_save(self, tmp_path, sample_spark_streaming_df):
+        filepath_json = (tmp_path / "test_streams").as_posix()
+        filepath_output = (tmp_path / "test_streams_output").as_posix()
+        schema_path = (tmp_path / SCHEMA_FILE_NAME).as_posix()
+        checkpoint_path = (tmp_path / "checkpoint").as_posix()
+
+        spark_json_ds = SparkDataSet(
+            filepath=filepath_json,
+            file_format="json",
+            save_args=[{"mode", "overwrite"}],
```

Comment:
Should there be a test for other mode or is `overwrite` the only supported mode?

Reply:
This functions is to write a sample json file for following use, which is being used for creating a streaming dataframe. The main focus in this test should be with `StreamingDataSet` I think

Reply:
Overwrite is not the only supported mode, it supports all spark streaming mode, We tested the append mode during our POC.

Reply:
Should we have a separate test for other mode as well? this one is for `SparkDataSet`, not for `StreamingDataSet`, in this test we are testing `append` mode indeed, but we can check `complete` and `Update` modes too